datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -656,13 +658,13 @@ impl LogicalPlanBuilder {
         let mut join_on: Vec<(Expr, Expr)> = vec![];
         let mut filters: Option<Expr> = None;
         for (l, r) in &on {
-            if self.plan.schema().field_from_column(l).is_ok()
-                && right.schema().field_from_column(r).is_ok()
+            if self.plan.schema().has_column(l)?
```

Comment:
propagating the error via `?` is not correct (this isn't an error, it is check, I think)